src/main/java/org/opentripplanner/routing/api/request/request/filter/TransitFilterRequest.java

```diff
@@ -0,0 +1,133 @@
+package org.opentripplanner.routing.api.request.request.filter;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.opentripplanner.framework.tostring.ToStringBuilder;
+import org.opentripplanner.transit.model.network.Route;
+import org.opentripplanner.transit.model.timetable.TripTimes;
+
+public class TransitFilterRequest implements Serializable, TransitFilter {
+
+  public static class Builder {
+
+    private List<SelectRequest> select = new ArrayList<>();
+    private List<SelectRequest> not = new ArrayList<>();
+
+    public Builder addSelect(SelectRequest selectRequest) {
+      this.select.add(selectRequest);
+
+      return this;
+    }
+
+    public Builder addNot(SelectRequest selectRequest) {
+      this.not.add(selectRequest);
+
+      return this;
+    }
+
+    public TransitFilterRequest build() {
+      return new TransitFilterRequest(this);
+    }
+  }
+
+  public TransitFilterRequest(Builder builder) {
+    this.select = Collections.unmodifiableList(builder.select);
```

Comment:
```suggestion
    this.select = List.of(builder.select);
```

I don't think it makes much of a difference, but it is shorter, that's why I prefer it.

Reply:
It makes a difference, the `unmodifiableList ` is a view, the `List.of` is a read-only copy. The `List.of` is preferred, if `Collections.unmodifiableList`  it is most likely that you should document why - e.g. performance, see updates ...

Reply:
In this instance you can't use List.of() since builder.select is a list. We could do builder.select.stream().toList() but Collections.unmodifiableList is shorter. Here it does not really matter if it's a view or copy.

Reply:
You don't need to change it but there is also `List.copyOf`.